src/FSharp.Core/prim-types.fs

```diff
@@ -4771,17 +4771,17 @@ namespace Microsoft.FSharp.Core
             [<CompiledName("Compare")>]
             let inline compare (e1: ^T) (e2: ^T) : int = 
                  (if e1 < e2 then -1 elif e1 > e2 then 1 else 0)
-                 when ^T : bool   = if (# "clt" e1 e2 : bool #) then (-1) else (# "cgt" e1 e2 : int #)
-                 when ^T : sbyte  = if (# "clt" e1 e2 : bool #) then (-1) else (# "cgt" e1 e2 : int #)
-                 when ^T : int16  = if (# "clt" e1 e2 : bool #) then (-1) else (# "cgt" e1 e2 : int #)
-                 when ^T : int32  = if (# "clt" e1 e2 : bool #) then (-1) else (# "cgt" e1 e2 : int #)
-                 when ^T : int64  = if (# "clt" e1 e2 : bool #) then (-1) else (# "cgt" e1 e2 : int #)
-                 when ^T : nativeint  = if (# "clt" e1 e2 : bool #) then (-1) else (# "cgt" e1 e2 : int #)
-                 when ^T : byte   = if (# "clt.un" e1 e2 : bool #) then (-1) else (# "cgt.un" e1 e2 : int #)
-                 when ^T : uint16 = if (# "clt.un" e1 e2 : bool #) then (-1) else (# "cgt.un" e1 e2 : int #)
-                 when ^T : uint32 = if (# "clt.un" e1 e2 : bool #) then (-1) else (# "cgt.un" e1 e2 : int #)
-                 when ^T : uint64 = if (# "clt.un" e1 e2 : bool #) then (-1) else (# "cgt.un" e1 e2 : int #)
-                 when ^T : unativeint = if (# "clt.un" e1 e2 : bool #) then (-1) else (# "cgt.un" e1 e2 : int #)
+                 when ^T : bool   = (# "cgt" e1 e2 : int #) - (# "clt" e1 e2 : int #)
```

Comment:
I think the `cgt` and `clt` are unnecessary as `bool` is spec'd as 0 or 1

Reply:
True. 😁

Reply:
I guess you still have to get 0, 1, - 1 somehow :) There must be some magic to do that. Maybe cgt/clt is good enough

Reply:
In IL True is 1 and False is 0 integer
Son
True > False 1 - 0 = 1
False < True 0 - 1 = - 1
True = True 1-1 = 0
False = False 0-0 = 0

So a simple subtraction x-y is enough